internal/oauthtoken/cache_alert.go

```diff
@@ -27,24 +27,24 @@ import (
 // a different cluster is changing, and produces a warning over an appropriate
 // communication channel.
 type CacheAlert struct {
-	impl   LoadStorer
+	LoadStorer
 	stderr io.Writer
 }
 
 func NewCacheAlert(impl LoadStorer, stderr io.Writer) LoadStorer {
 	return &CacheAlert{
-		impl:   impl,
-		stderr: stderr,
+		LoadStorer: impl,
+		stderr:     stderr,
 	}
 }
 
 func (a *CacheAlert) Store(ctx context.Context, cluster string, token *oauth2.Token) error {
-	oldToken, err := a.impl.Load(ctx, cluster)
+	oldToken, err := a.Load(ctx, cluster)
```

Comment:
nit: This is slightly fewer lines of code, which I do appreciate, but seems slightly more complicated:
* there used to be only one way to delegate calls: `a.impl.Xxx`. Now there are two: `a.LoadStorer.Xxx` and `a.Xxx` (iff `a.Xxx` isn't overridden)
* the code happens to be equivalent to the previous version today, but if `a.Load` is overridden in the future, the behavior of `a.Store` will change (perhaps surprisingly) in this version, whereas it wouldn't have in the old version

I think I still have a (slight/non-blocking) preference for the previous version - but I'd also like to know why you prefer this way!

Reply:
`CacheAlert` is a decorator type, and I think embedding is actually ideal here because we only need to override the methods we actually want to decorate. If we add more methods to the base type (like `Delete`), we don't need to change anything here.

About two ways to call:

- Call `a.LoadStorer.Store` only from `a.Store` to call the underlying method. It's like calling `super.someMethod` in languages with classes.
- Call `a.Store` in all other cases.

That being said, I usually prefer not to use decorators because they increase indirection, making the code harder to navigate. I think it'd be better to to pull this code out into the call site in `main.go` or perhaps a separate wrapper function. I'm trying to avoid changing too much though.